fuel-core-interfaces/src/bft.rs

```diff
@@ -0,0 +1,17 @@
+use super::model::{FuelBlock, SealedFuelBlock};
+use anyhow::Result;
+use std::sync::Arc;
+use tokio::sync::oneshot;
+
+pub enum BFTMpsc {
```

Comment:
I don't think we should define the specifics of the behavior in the consumer level.

Instead, we can use [fuel_bft::Message](https://github.com/FuelLabs/fuel-bft/blob/7acc238ee02ec8cbc726deb97519b075fc0ab538/src/reactor/message.rs#L11-L20) so we inherit the expected behavior of the consensus implementation.

The proposed approach will lead to duplicated logic and restriction of functionalities since this interface cannot be aware of the capabilities of fuel-bft.

As example, the variant `BFTMpsc::Stop` is supersede by [Notification::Kill](https://github.com/FuelLabs/fuel-bft/blob/7acc238ee02ec8cbc726deb97519b075fc0ab538/src/reactor/message/notification.rs#L9-L10)

Defining the structures here might lead to either duplication or circular dependencies since either fuel-bft will reimplement these variants or will require fuel-core

Reply:
`fuel-bft` has a very opinionated interface, but in `fuel-core` it is just 1 out of 7/8 modules.
This is binding that I talked about in chat, and it can be done inside `fuel-core-bft`.

Reply:
I think it's fine for fuel-core to define it's own messages between components, and wrap / adapt the message interfaces defined in fuel-bft.

Reply:
I think it's fine for fuel-core to define it's own messages between components, and wrap / adapt the message interfaces defined in fuel-bft.